### DISCUSSION TITLE
Enforce certain params in non-grid-search sweeps?

### DISCUSSION THREAD

**Discussion by colobas** [facebookresearch/hydra#2809] [Q&A] [answered]
I'm wondering if there's a way to make sure that certain parameters are tried out in sweeps. This may be more of an Optuna/Ax question, and it might be less general than I'm currently thinking but:

- say I'm doing hyperparameter search over a certain search space
- say one of my config's parameters is `seed`, and I want to do a sweep over different seeds
- is there a way to ensure that each configuration proposed by the hyperparam search gets tried out with all of the seeds?

**Comment by odelalleau** [comment, marked as answer]
I think you'd need to implement your own sweeper that would be a wrapper on top of an existing sweeper to automatically "expand" the sweep space the way you want (through this new sweeper config). I'm not super familiar with plugins though, so there might be another way...

**Reply by colobas** [reply]
Yeah that sounds about right, it's what I expected. Thanks for the quick reply!
